src/movies.js

```diff
@@ -1,25 +1,119 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+	return moviesArray.map((movieObj) => {
+		return movieObj.director;
+	});
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+	let spielbergDramas = moviesArray.filter((movieObj) => {
+		return (
+			movieObj.genre.includes("Drama") &&
+			movieObj.director === "Steven Spielberg"
+		);
+	});
+
+	return spielbergDramas.length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+	if (moviesArray.length === 0) {
+		return 0;
+	} else {
+		let scoreSum =
+			moviesArray.reduce((acc, curr) => {
+				if (typeof curr.score !== "number") {
+					curr.score = 0;
+					return acc + curr.score;
+				} else {
```

Comment:
You could remove the `else` statement because you're doing the same thing inside the `if `and the `else `with the `return`, so you could only change the score inside the `if `and outside the `if `you call the `return`
```
	moviesArray.reduce((acc, curr) => {
				if (typeof curr.score !== 'number') {
					curr.score = 0
				}
				return acc + curr.score
			}, 0) / moviesArray.length
```